### DISCUSSION TITLE
Order the parent results by data from a one-to-many relationship

### DISCUSSION THREAD

**Discussion by psaxde** [prisma/prisma#15127] [Q&A] [answered]
Hello 👋

Given two models:

```
**Product**
Product.id

**ProductMetricsLatest**
ProductMetricsLatest.id
ProductMetricsLatest.productId
ProductMetricsLatest.channelId
ProductMetricsLatest.pageViewsSevenDays
```

How can something like this SQL be accomplished with Prisma? Is $queryRaw the only way to do that?

```
SELECT
	products.id,
	products.product_number,
	product_metrics_latest.page_views_seven_days
FROM
	products
	LEFT JOIN product_metrics_latest ON product_metrics_latest.product_id = products.id
		AND product_metrics_latest.channel_id = 2
WHERE
	product_metrics_latest.id IS NOT NULL
ORDER BY
	product_metrics_latest.page_views_seven_days DESC;
```

Doing something like this does not work:

```
include: {
  productMetricLatest: {
    orderBy: {
      pageViewsSevenDays: 'desc',
    },
  },
},
```

The created SQL for the SELECT * FROM products does not contain any order statement. The SELECT * FROM product_metrics_latest does, but as mentioned I do want to sort the parent data.

On the first orderBy level I only have the ability to sort by aggregated value count of ProductMetricsLatest Relationship.

As you can see above, the ProductMetricsLatest Model must be a one-to-many relationship because of we've some additional criteria/field (channel_id) here. 

I do not want to sort the data of the relation (ProductMetricsLatest), I want to sort the Parent (Products) by a relation.

Cheers

Ben

**Comment by austincrim** [comment, marked as answer]
Hey @psaxde 👋,

I think you will need to stick with [`$queryRaw`](https://www.prisma.io/docs/concepts/components/prisma-client/raw-database-access#queryraw) for now.

Feel free to open a [feature request](https://github.com/prisma/prisma/issues/new?assignees=&labels=&template=feature_request.md&title=) if you have thoughts on how we could support this kind of pattern in the future.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
